provider/resources.go

```diff
@@ -2190,6 +2191,68 @@ func Provider() tfbridge.ProviderInfo {
+					{Value: "ap-east-1", Name: "APEast1"},
+					{Value: "ap-northeast-1", Name: "APNortheast1"},
+					{Value: "ap-northeast-2", Name: "APNortheast2"},
+					{Value: "ap-south-1", Name: "APSouth1"},
+					{Value: "ap-southeast-2", Name: "APSoutheast2"},
+					{Value: "ap-southeast-1", Name: "APSoutheast1"},
+					{Value: "ca-central-1", Name: "CACentral"},
+					{Value: "cn-north-1", Name: "CNNorth1"},
+					{Value: "cn-northwest-1", Name: "CNNorthwest1"},
+					{Value: "eu-central-1", Name: "EUCentral1"},
+					{Value: "eu-north-1", Name: "EUNorth1"},
+					{Value: "eu-west-1", Name: "EUWest1"},
+					{Value: "eu-west-2", Name: "EUWest2"},
+					{Value: "eu-west-3", Name: "EUWest3"},
+					{Value: "eu-south-1", Name: "EUSouth1"},
+					{Value: "me-south-1", Name: "MESouth1"},
+					{Value: "sa-east-1", Name: "SAEast1"},
+					{Value: "us-gov-east-1", Name: "USGovEast1"},
+					{Value: "us-gov-west-1", Name: "USGovWest1"},
+					{Value: "us-east-1", Name: "USEast1"},
+					{Value: "us-east-2", Name: "USEast2"},
+					{Value: "us-west-1", Name: "USWest1"},
+					{Value: "us-west-2", Name: "USWest2"},
+				},
+			},
+			"aws:lambda/Runtime:Runtime": {
+				ObjectTypeSpec: schema.ObjectTypeSpec{
+					Type: "string",
+				},
+				Enum: []*schema.EnumValueSpec{
+					{Value: "dotnetcore2.1", Name: "DotnetCore2d1"},
+					{Value: "dotnetcore3.1", Name: "DotnetCore3d1"},
+					{Value: "go1.x", Name: "Go1dx"},
+					{Value: "java8", Name: "Java8"},
+					{Value: "java11", Name: "Java11"},
+					{Value: "ruby2.5", Name: "Ruby2d5"},
+					{Value: "ruby2.7", Name: "Ruby2d7"},
+					{Value: "nodejs10.x", Name: "NodeJS10dX"},
+					{Value: "nodejs12.x", Name: "NodeJS12dX"},
+					{Value: "python2.7", Name: "Python2d7"},
+					{Value: "python3.6", Name: "Python3d6"},
+					{Value: "python3.7", Name: "Python3d7"},
+					{Value: "python3.8", Name: "Python3d8"},
+					{Value: "provided", Name: "Custom"},
+					{Value: "nodejs", Name: "NodeJS", DeprecationMessage: "No longer supported. New lambda functions created using this runtime will fail."},
+					{Value: "nodejs4.3-edge", Name: "NodeJS4d3Edge", DeprecationMessage: "No longer supported. New lambda functions created using this runtime will fail."},
+					{Value: "nodejs4.3", Name: "NodeJS4d3", DeprecationMessage: "No longer supported. New lambda functions created using this runtime will fail."},
+					{Value: "nodejs6.10", Name: "NodeJS6d10", DeprecationMessage: "No longer supported. New lambda functions created using this runtime will fail."},
+					{Value: "nodejs8.10", Name: "NodeJS8d10", DeprecationMessage: "No longer supported. New lambda functions created using this runtime will fail."},
+					{Value: "dotnetcore1.0", Name: "DotnetCore1d0", DeprecationMessage: "No longer supported. New lambda functions created using this runtime will fail."},
+					{Value: "dotnetcore2.0", Name: "DotnetCore2d0", DeprecationMessage: "No longer supported. New lambda functions created using this runtime will fail."},
```

Comment:
Do we need to carry these deprecated values forward in the schema? Seems like a good opportunity to avoid adding new deprecated values to the new enums. Of course, we'd keep the old deprecated constants in the overlays.

Reply:
Another thing: Should we do a pass over the current list of non-deprecated values to see if they should be deprecated (and if so, deprecate the overlay constant, and not add them here)?

Reply:
Good call, will do.

Reply:
@justinvp we can't exclude them from the new enums if we still want to maintain the older constants. The reason is that codegen is now responsible for creating the union type that determines all possible values i.e. `type SomeType = "foo" | "bar`. If we exclude those values from the schema, we can't have constants of the enum type that declare values outside of the type i.e. `const BazSomeType: SomeType = "baz"`

Reply:
Ah, bummer.

Reply:
I did find some new ones though and only added them to the schematized enums.

Reply:
Was thinking about this more, and I think we could still remove these from the schema (and new enum), and still maintain the old constants.

Instead of having these defined as follows (as they currently are):

```typescript
/** @deprecated No longer supported. New lambda functions created using this runtime will fail */
export let NodeJSRuntime: Runtime = "nodejs";
```

We change it to:

```typescript
/** @deprecated No longer supported. New lambda functions created using this runtime will fail */
export let NodeJSRuntime = "nodejs" as Runtime;
```

The `as Runtime` [type assertion](https://www.typescriptlang.org/docs/handbook/basic-types.html#type-assertions) makes the TS compiler happy, even though "nodejs" is not really a valid value of the new `Runtime` enum.

I couldn't come up with any downsides.

Reply:
Kinda feels like cheating 😅